src/kboot.c

```diff
@@ -327,6 +327,49 @@ static int dt_set_wifi(void)
     return 0;
 }
 
+static void dt_set_uboot_dm_preloc(int node)
+{
+    // Tell U-Boot to bind this node early
+    fdt_setprop_empty(dt, node, "u-boot,dm-pre-reloc");
+
+    // Make sure the power domains are bound early as well
+    int pds_size;
+    const fdt32_t *pds = fdt_getprop(dt, node, "power-domains", &pds_size);
+    if (pds) {
```

Comment:
I think this won't work for >1 PD, since the FDT mutations will invalidate the pointer. You'll have to copy it out before iterating.

Reply:
Right.  The device tree I tested this on only has single PD dependencies.  Pushed a fix onto the branch.